lib/flux/app-state.js

```diff
@@ -136,7 +136,9 @@ export const actionMap = new ActionMap({
     newTag: {
       creator({ tagBucket, name }) {
         return () => {
-          tagBucket.add({ name }, () => {
+          // tag.id must be the tag name in lower case and percent escaped
+          const tagId = encodeURIComponent(name.toLowerCase());
```

Comment:
is this how the mobile clients create them or are they using a kind of UUID? not sure if we have reason to match them up with mobile clients but I'm pretty sure that the tag id is intentionally unrelated to the label

someone creates a label, adds it to a bunch of notes, decides the label should be something else (e.g. "work" to "naperville file"), renames the label, now that tag has nothing to do with the text "work"

Reply:
Yeah it doesn't make much sense after you rename a tag. But it is indeed how the other apps are doing it: https://github.com/Automattic/simplenote-macos/blob/develop/Simplenote/TagListViewController.m#L237